cmd/istioctl/main.go

```diff
@@ -491,7 +491,7 @@ func init() {
 	rootCmd.PersistentFlags().StringVarP(&istioNamespace, "istioNamespace", "i", kube.IstioNamespace,
 		"Istio system namespace")
 
-	rootCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", v1.NamespaceDefault,
+	rootCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", getDefaultNamespace(kubeconfig),
```

Comment:
Closer :)

`namespace` should be set _after_ flags are parsed instead of when they are defined. Specifically, add the following code to  `rootCmd.PersistentPreRun` 

```golang
if namespace == "" {
    namespace = getDefaultNamespace(kubeconfig)
}
```

Reply:
ok, thank you for the additional information. I've never used cobra before but I think I get it now.